packages/salesforce/src/Adaptor.js

```diff
@@ -110,17 +110,18 @@ export function bulk(sObjectName, operation, records, options = {}) {
       pollInterval = 6000,
     } = resolvedOptions;
 
-    if (allowNoOp && resolvedRecords.length === 0) {
+    const flattenRecords = util.flattenData(resolvedRecords);
```

Comment:
This not good English (because flatten is a verb form and you should use an adjective here ) - I would rename the variable `flatRecords` I  think.